packages/x-components/src/x-modules/facets-next/components/facets/__tests__/facets-provider.spec.ts

```diff
@@ -0,0 +1,175 @@
+
+describe('testing Facets component', () => {
+  it('is an XComponent', () => {
+    const { wrapper } = renderFacetsProviderComponent();
+    expect(isXComponent(wrapper.vm)).toEqual(true);
+  });
+
+  it('has FacetsModule as XModule', () => {
+    const { wrapper } = renderFacetsProviderComponent();
+    expect(getXComponentXModuleName(wrapper.vm)).toEqual('facetsNext');
+  });
+
+  it('keeps the state facets if no facets are provided', () => {
+    const stateFacets = {
+      color_facet: createSimpleFacetStub('color', createSimpleFilter => [
+        createSimpleFilter('Red', false),
+        createSimpleFilter('Blue', true)
+      ]),
+      brand_facet: createSimpleFacetStub('brand', createSimpleFilter => [
+        createSimpleFilter('Adidas', false),
+        createSimpleFilter('Nike', false)
+      ])
+    };
+    const { getStateFacets } = renderFacetsProviderComponent({
+      stateFacets,
+      providedFacets: undefined
+    });
+    expect(getStateFacets()).toEqual(stateFacets);
+  });
+
+  it('renders the provided facets keeping the state facets', () => {
+    const stateFacets = {
+      color_facet: createSimpleFacetStub('color', createSimpleFilter => [
+        createSimpleFilter('Red', false),
+        createSimpleFilter('Blue', true)
+      ]),
+      brand_facet: createSimpleFacetStub('brand', createSimpleFilter => [
+        createSimpleFilter('Adidas', false),
+        createSimpleFilter('Nike', false)
+      ])
+    };
+    const providedFacets: Facet[] = [
+      createSimpleFacetStub('size', createSimpleFilter => [
+        createSimpleFilter('Small', false),
+        createSimpleFilter('Large', true)
+      ]),
+      createSimpleFacetStub('stock', createSimpleFilter => [
+        createSimpleFilter('no stock', false),
+        createSimpleFilter('with stock', false)
+      ])
+    ];
+    const { getStateFacets } = renderFacetsProviderComponent({
+      stateFacets,
+      providedFacets
+    });
+
+    expect(getStateFacets()).toEqual(
+      Object.assign(stateFacets, arrayToObject(providedFacets, 'id'))
+    );
+  });
+
+  it('emits the `UserChangedSelectedFilters` event when a filter is changed', async () => {
+    const { wrapper, deselectFilters, selectFilters } = renderFacetsProviderComponent();
+
+    const colorFacet = createSimpleFacetStub('color', createSimpleFilter => [
+      createSimpleFilter('Red', false),
+      createSimpleFilter('Blue', false)
+    ]);
+    const sizeFacet = createSimpleFacetStub('size', createSimpleFilter => [
+      createSimpleFilter('Big', false),
+      createSimpleFilter('Small', true)
+    ]);
+    const [redFilter] = colorFacet.filters;
+    const [bigFilter, smallFilter] = sizeFacet.filters;
+
+    // Setting facets with selected filters does not trigger UserChangedSelectedFilters
+    await wrapper.setProps({ facets: [colorFacet, sizeFacet] });
+    expect(wrapper.emitted('UserChangedSelectedFilters')).toBeUndefined();
```

Comment:
I would keep the event payload as a check. Maybe we can use the `areFiltersDifferent` to keep the check simple